ksqldb-execution/src/main/java/io/confluent/ksql/execution/function/UdafUtil.java

```diff
@@ -49,44 +52,89 @@ private UdafUtil() {
+              .getFunction(args);
+
+      final int totalArgs = functionCall.getArguments().size();
+
+      // All non-constant UDAF arguments must be column references
+      final List<Integer> argIndices = functionCall.getArguments().stream()
+              .limit(totalArgs - func.initArgs)
+              .map((arg) -> {
+                final Optional<Column> column;
+
+                if (arg instanceof UnqualifiedColumnReferenceExp) {
+                  final UnqualifiedColumnReferenceExp colRef =
+                          (UnqualifiedColumnReferenceExp) arg;
+                  column = schema.findValueColumn(colRef.getColumnName());
+                } else {
+                  // assume that it is a column reference with no alias
+                  column = schema.findValueColumn(ColumnName.of(arg.toString()));
+                }
+
+                return column.orElseThrow(
+                        () -> new KsqlException("Could not find column for expression: " + arg)
+                );
+              }).map(Column::index).collect(Collectors.toList());
+
+      return func.source.apply(createAggregateFunctionInitArgs(
+              func.initArgs,
+              argIndices,
+              functionCall,
+              config
+      ));
     } catch (final Exception e) {
       throw new KsqlException("Failed to create aggregate function: " + functionCall, e);
     }
   }
 
+  /**
+   * Creates the initial arguments for a dummy aggregate function that will not
+   * actually be called. For example, this is useful for retrieving the return type
+   * in the {@link ExpressionTypeManager}.
+   * @param numRegularArgs  number of regular, column-dependent arguments. It is not
+   *                        always possible to compute this from the number of initial
+   *                        arguments and the number of arguments to the function call
+   *                        in case a default argument is provided when the user does
+   *                        not provided any of their own arguments.
+   * @param numInitArgs     number of initial arguments. This is necessary as there
+   *                        is the potential for the {@link FunctionRegistry} to
+   *                        resolve different functions based on the number of initial
+   *                        arguments.
+   * @param functionCall    the call to the function whose initial arguments should be
+   *                        created
+   * @return initial arguments for a dummy aggregate function
+   */
   public static AggregateFunctionInitArguments createAggregateFunctionInitArgs(
-      final int udafIndex,
-      final FunctionCall functionCall,
-      final KsqlConfig config
+          final int numRegularArgs,
+          final int numInitArgs,
+          final FunctionCall functionCall
+  ) {
+    return createAggregateFunctionInitArgs(
+            numInitArgs,
+            IntStream.range(0, numRegularArgs).boxed().collect(Collectors.toList()),
```

Comment:
Since this is a dummy function, this can be an empty list right?

Reply:
I originally used an empty list here, but it caused an issue, so I created a list with the same length as the number of regular args. I don't think this is still necessary with other changes, but I will check to make sure.